main/main.c

```diff
@@ -172,22 +186,32 @@ gatekeeper_log_init(void)
 		return -1;
 	}
 
-	new_log_file = fopen(log_file_path, "a");
+	log_fd = open(log_file_path, O_CREAT, log_file_mode);
+	if (log_fd < 0) {
+		G_LOG(ERR, "gatekeeper: Failed to get log file descriptor %s - %s\n",
+			log_file_path, strerror(errno));
+		return -1;
+	}
+
+	new_log_file = fdopen(log_fd, "rw");
```

Comment:
Shouldn't the mode should be `"a"` to avoid losing the content of previously written log file when the log file being open has the same name of a previously written log file?

I couldn't be sure based on the manula, but we may need to need to add `O_APPEND` to the flags of `open()` as well.

Reply:
Without ``O_APPEND``, it also works as expected.

Reply:
Great. Let's **not** add `O_APPEND` in this case.